File: miniJobs/Domain/Entities/Rating.cs
using Domain.Common;
using Domain.Interfaces;

namespace Domain.Entities;

[Table("ratings")]
public class Rating : BaseAuditableEntity, IEntity<int>
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("value")]
    public int Value { get; set; }

    [Column("comment")]
    public string Comment { get; set; }

    [Column("job_application_id")]
    public int JobApplicationId { get; set; }

    [Column("rated_user_id")]
    public int RatedUserId { get; set; }

    [Column("is_active")]
    public bool IsActive { get; set; }

    [NotMapped]
    public string CreatedByFullName { get; set; }

    [NotMapped]
    public string RatedUserFullName { get; set; }

    [NotMapped]
    public string CreatedByRole{ get; set; }

    [NotMapped]
    public string RatedUserRole { get; set; }

    [NotMapped]
    public byte[]? Photo { get; set; }

}

File: miniJobs/UI/minijobs_admin/lib/models/rating.dart
import 'dart:convert';
import 'dart:typed_data';

import 'package:json_annotation/json_annotation.dart';

part 'rating.g.dart';

@JsonSerializable()
class Rating {
   int id;
   String comment;
   int value;
   int jobApplicationId;
   int ratedUserId;
   String createdByFullName;
   DateTime created;
   Uint8List? photo;
   int createdBy;
   String createdByRole;
   String ratedUserFullName;
   String ratedUserRole;
   bool isActive;

  Rating(
     this.id,
     this.comment,
     this.value,
     this.jobApplicationId,
     this.ratedUserId,
     this.createdByFullName,
     this.created,
      this.photo,
      this.createdBy,
      this.createdByRole,
      this.ratedUserFullName,
      this.ratedUserRole,
      this.isActive
  );

  factory Rating.fromJson(Map<String, dynamic> json) => _$RatingFromJson(json);

  Map<String, dynamic> toJson() => _$RatingToJson(this);
}
